contracts/poap-manager/src/contract.rs

```diff
@@ -0,0 +1,530 @@
+        if config.admin != info.sender {
+            return Err(ContractError::NotAdmin {
+                caller: info.sender.clone(),
+            });
+        }
+        config.admin = new_admin.clone();
+        Ok(config)
+    })?;
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_UPDATE_ADMIN)
+        .add_attribute(ATTRIBUTE_NEW_ADMIN, new_admin)
+        .add_attribute(ATTRIBUTE_SENDER, info.sender))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps<DesmosQuery>, _env: Env, msg: QueryMsg) -> StdResult<QueryResponse> {
+    match msg {
+        QueryMsg::Config {} => to_binary(&query_config(deps)?),
+    }
+}
+
+fn query_config(deps: Deps<DesmosQuery>) -> StdResult<QueryConfigResponse> {
+    let config = CONFIG.load(deps.storage)?;
+    Ok(QueryConfigResponse {
+        admin: config.admin,
+        poap_code_id: config.poap_code_id,
+        poap_contract_address: POAP_CONTRACT_ADDRESS.load(deps.storage)?,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use cosmwasm_std::testing::{mock_env, mock_info};
+    use cosmwasm_std::{StdError, SubMsgResponse, SubMsgResult, Timestamp};
+    use cw721_base::InstantiateMsg as Cw721InstantiateMsg;
+    use cw_utils::ParseReplyError;
+    use desmos_bindings::mocks::mock_queriers::mock_dependencies_with_custom_querier;
+    use poap::msg::{EventInfo, InstantiateMsg as POAPInstantiateMsg};
+    use prost::Message;
+
+    const CREATOR: &str = "desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc";
+    const NEW_ADMIN: &str = "desmos1fcrca0eyvj32yeqwyqgs245gjmq4ee9vjjdlnz";
+
+    fn get_valid_instantiate() -> InstantiateMsg {
+        InstantiateMsg {
+            admin: "desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc".into(),
+            poap_code_id: 1u64.into(),
+            poap_instantiate_msg: POAPInstantiateMsg {
+                admin: CREATOR.into(),
+                minter: CREATOR.into(),
+                cw721_code_id: 2u64.into(),
+                cw721_initiate_msg: Cw721InstantiateMsg {
+                    minter: CREATOR.into(),
+                    name: CREATOR.into(),
+                    symbol: CREATOR.into(),
+                },
+                event_info: EventInfo {
+                    creator: CREATOR.into(),
+                    start_time: Timestamp::from_seconds(10),
+                    end_time: Timestamp::from_seconds(20),
+                    per_address_limit: 2,
+                    base_poap_uri: "ipfs://popap-uri".to_string(),
+                    event_uri: "ipfs://event-uri".to_string(),
+                },
+            },
+        }
+    }
+
+    #[derive(Clone, PartialEq, Message)]
+    struct MsgInstantiateContractResponse {
+        #[prost(string, tag = "1")]
+        pub contract_address: ::prost::alloc::string::String,
+        #[prost(bytes, tag = "2")]
+        pub data: ::prost::alloc::vec::Vec<u8>,
+    }
+
+    fn get_valid_instantiate_reply() -> Reply {
```

Comment:
We should avoid to have a custom serializer for an internal `cosmwasm-std` struct, if the cosmwasm team changes this struct our tests will probably start to fail. We should instead relay on the integration tests to check that the contract have the correct state with an integratino test.